inject-generator/src/main/java/io/avaje/inject/generator/TypeExtendsReader.java

```diff
@@ -148,59 +155,73 @@ void process(boolean forBean) {
   }
 
   private String initProvidesAspect() {
-    for (String providesType : providesTypes) {
+    for (final String providesType : providesTypes) {
       if (Util.isAspectProvider(providesType)) {
         return Util.extractAspectType(providesType);
       }
     }
     return "";
   }
 
-  private void addSuperType(TypeElement element) {
+  private void addSuperType(TypeElement element, TypeMirror mirror) {
     readInterfaces(element);
-    final String fullName = element.getQualifiedName().toString();
-    if (!fullName.equals(JAVA_LANG_OBJECT) && !fullName.equals(JAVA_LANG_RECORD)) {
+    final String fullName = mirror.toString();
```

Comment:
What is the thinking behind this change to use the mirror.toString() ?

Reply:
Ah so `element.getQualifiedName().toString()` loses the generic type info ...

Reply:
exactly